rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala

```diff
@@ -148,9 +171,31 @@ private class Translator {
         () =>
           binaryExpr(toExpr(x), Const(a.toDouble), ring.times)
     }
-    combineTree(lazyExpr, ring)
   }
 
+  private def combineTerms(terms: Seq[(Real, BigDecimal)], ring: Ring): Expr = {
+    val lazyExprs = makeLazyExprs(terms, ring)
+    combineTree(lazyExprs, ring)
+  }
+
+  private def combineSumTerms(terms: Seq[(Real, BigDecimal)]): Expr = {
+    val ring = multiplyRing
+    val lazyExprs = makeLazyExprs(terms, ring)
+    foldChain(lazyExprs, Const(0.0), 0)
```

Comment:
We don't actually care about the initial value of the accumulator: we keep track of which iteration we are on and replace it with the first term in the list on iteration 0.

Reply:
Weird :) Is it necessary? And why can't we just use a normal foldLeft?

Reply:
Could this be 

```scala
lazyExprs.tail.foldLeft(lazyExprs.head){
(acc, t) => binaryExpr(acc, t(), AddOp)
}
```

Reply:
yes, definitely :)